feat(signals): allow defining private SignalStore members (#4447)
ngrx/platform: projects/ngrx.io/content/guide/signals/signal-store/entity-management.md

);

</code-example>

## Private Entity Collections

> Private entity collection can be defined by using the `_` prefix for the collection name.

Reviewer: nit: phrasing

```suggestion
Private entity collections are defined by using the `_` prefix for the collection name.
```

```ts
const todoConfig = entityConfig({
  entity: type<Todo>(),
  // 👇 private collection

>   collection: '_todo',

Reviewer 2: Should we add a JSDoc comments for the `collection` property to mention this behavior?
I'm thinking that new devs wouldn't expect this behavior by naming a collection.

Author: Definitely! I'm going to add JSDoc for all public APIs.

});

const TodosStore = signalStore(
  withEntities(todoConfig),
  withComputed(({ _todoEntities }) => ({
    // 👇 exposing entity array publicly
    todos: _todoEntities,
  }))
);

@Component({
  /* ... */
  template: `
    <h1>Todos</h1>
    <ngrx-todo-list [todos]="store.todos()" />
  `,
  providers: [TodosStore],
})
class TodosComponent {
  readonly store = inject(TodosStore);
}
```

<div class="alert is-helpful">

> Learn more about private store members on [the following page](/guide/signals/signal-store/private-store-members).

Reviewer: ```suggestion
Learn more about private store members in the [private store members guide](/guide/signals/signal-store/private-store-members).
```
